### ISSUE TITLE
Improve transaction selection 

### ISSUE THREAD

**Issue by justbudget** [justbudget/justbudget#66]
1. When selecting an individual transaction, any other currently selected transactions should unselect.
1. When hitting `Shift` + `Click`, I should be able to select all transactions continuously between the currently selected transaction. 
1. When hitting `Ctrl` + `Click`, I should be able to select multiple non-continuous transactions.

**Comment by bpaulien** [comment]
I agree with you about 2 and 3. I would be highly annoyed if 1 were implemented.

**Comment by Rick-Thomas** [comment]
I think I'm to blame for that suggestion!!!    

I don't know if you saw my comments here:  https://www.reddit.com/r/justbudget/comments/vg964r/transaction_selection_suggestion/?utm_source=share&utm_medium=web2x&context=3   

I don't know what your use case is...   Could it be similar to the option 3???

**Comment by justbudget** [comment]
@bpaulien good to know! I could see both sides on 1. When I get around to this, maybe I'll start with 2 and 3.

**Comment by bpaulien** [comment]
Yeah, maybe make it an option in settings if people actually want that functionality? I can't imagine ever finding it helpful. (speaking, of course, about option 1)

**Comment by Rick-Thomas** [comment]
JustBudget today works as alternative 3...   as I go clicking on transaction, contiguous or not, the get selected...

My suggestion is based on windows and other apps "method" (if we can call it that!!) of selection...

If the recommendation is not to go with 1, as @bpaulien suggests, then I would scratch 3 also, because that happens today...

When I have several selected and want to change the selection for new ones, I'd have to **a)** go to the box at the top left of the titles to deselect and then **b)** start new selection...   with method 1, just by selecting a new one would start the new selection process...   Which, bottom line, is the same in a one-click vs a two-click "process"...

So, to sum it up, I'd only concentrate on 2, which will help us avoid having to clicking several contiguous transaction...   The other (1 & 3) go together and save one click!!!   Good discussion, it brought 3 requests down to 1!!!   ;-)

**Comment by bpaulien** [comment]
After reading the original post on reddit and reading through this and the resulting comments, I am slightly less against it... Just as long as it's obvious to the common user that you have to hold control down while selecting. As @Rick-Thomas stated above, it's already behaving that way. And I think I was imagining it on the account register. In that case I would hate to have everything unselected if I forgot to hold the control key.

On the budget page, it would be less frustration... Mostly because I don't typically select a bunch of categories at a time.

**Comment by Rick-Thomas** [comment]
Using JB I've just realized that in the Budget, if I select a Category, depending on how I select it, two things happen:

1. If I select clicking on the category **line**, it unselects all the categories that are selected and selects that category...
2. If I select clicking in the category **box**, on the left of the line, it adds that category to whatever categories are selected...

So I think it all boils down to implementing feature 2, shift-click for selecting contiguous transactions or categories...   The other two (1 and 3) are already in JB!!!

**Comment by jbmadsen2022** [comment]
> So I think it all boils down to implementing feature 2, shift-click for selecting contiguous transactions or categories... The other two (1 and 3) are already in JB!!!

Please note that selecting budget categories and transactions behave differently. In my opinion the budget is "correct" and transactions is "wrong". It is non-standard behavior that transactions stay selected when you single-click a different line. The right way is to single-select when you single-click.

If you want select multiple lines, either shift-click to select a range, or ctrl-click to add non-contiguous lines. An alternative to ctrl-click is to use the check boxes on the left which should always just select the individual box on or off. There is a selector at the top to clear all checkboxes.

This would be similar to just about every other piece of software I can think of. A single example could be Excel, but there are countless others.
